query-engine/connectors/sql-query-connector/src/database/js.rs

```diff
@@ -0,0 +1,161 @@
+    if QUERYABLE.set(driver).is_err() {
+        panic!("Cannot register driver twice");
+    }
+}
+
+pub struct Js {
+    connector: JsConnector,
+    connection_info: ConnectionInfo,
+    features: psl::PreviewFeatures,
+    psl_connector: psl::builtin_connectors::JsConnector,
+}
+
+fn get_connection_info(url: &str) -> connector::Result<ConnectionInfo> {
+    ConnectionInfo::from_url(url).map_err(|err| {
+        ConnectorError::from_kind(ErrorKind::InvalidDatabaseUrl {
+            details: err.to_string(),
+            url: url.to_string(),
+        })
+    })
+}
+
+#[async_trait]
+impl FromSource for Js {
+    async fn from_source(
+        source: &psl::Datasource,
+        url: &str,
+        features: psl::PreviewFeatures,
+    ) -> connector_interface::Result<Js> {
+        let psl_connector = source.active_connector.as_js_connector().unwrap_or_else(|| {
+            panic!(
+                "Connector for {} is not a JsConnector",
+                source.active_connector.provider_name()
+            )
+        });
+
+        let connector = registered_js_connector().unwrap().clone();
+        let connection_info = get_connection_info(url)?;
+
+        return Ok(Js {
+            connector: JsConnector { queryable: connector },
+            connection_info,
+            features: features.to_owned(),
+            psl_connector,
+        });
+    }
+}
+
+#[async_trait]
+impl Connector for Js {
+    async fn get_connection<'a>(&'a self) -> connector::Result<Box<dyn Connection + Send + Sync + 'static>> {
+        super::catch(self.connection_info.clone(), async move {
+            let sql_conn = SqlConnection::new(self.connector.clone(), &self.connection_info, self.features);
+            Ok(Box::new(sql_conn) as Box<dyn Connection + Send + Sync + 'static>)
+        })
+        .await
+    }
+
+    fn name(&self) -> &'static str {
+        self.psl_connector.name
+    }
+
+    fn should_retry_on_transient_error(&self) -> bool {
+        false
+    }
+}
+
+// TODO: miguelff: I haven´t found a better way to do this, yet... please continue reading.
+//
+// There is a bug in NAPI-rs by wich compiling a binary crate that links code using napi-rs
+// bindings breaks. We could have used a JsQueryable from the `js-connectors` crate directly, as the
+// `connection` field of a `Js` connector, but that will imply using napi-rs transitively, and break
+// the tests (which are compiled as binary creates)
+//
+// To avoid the problem above I separated interface from implementation, making JsConnector
+// independent on napi-rs. Initially, I tried having a field Arc<&dyn TransactionCabable> to hold
+// JsQueryable at runtime. I did this, because TransactionCapable is the trait bounds required to
+// create a value of  `SqlConnection` (see [SqlConnection::new])) to actually performt the queries.
+// using JSQueryable. However, this didn't work because TransactionCapable is not object safe.
+// (has Sized as a supertrait)
```

Comment:
Yes, I remember being bit by a very similar problem in the recent past with the same `Sized` trait